python/pyarrow/dataset.py

```diff
@@ -411,7 +421,14 @@ def _filesystem_dataset(source, schema=None, filesystem=None,
     partitioning = _ensure_partitioning(partitioning)
 
     if isinstance(source, (list, tuple)):
-        fs, paths_or_selector = _ensure_multiple_sources(source, filesystem)
+        if all(_is_path_like(elem) for elem in source):
+            fs, paths_or_selector = _ensure_multiple_sources(source,
+                                                             filesystem)
+        else:
+            fs, paths_or_selector = _MockFileSystem(), source
```

Comment:
Why the MockFileSystem fallback here?

Reply:
This is a dirty hack to get the ParquetDataset tests passing; it's ignored later except that `fs` must be a non-`None` FileSystem.